oak-search/src/main/java/org/apache/jackrabbit/oak/plugins/index/search/spi/query/FulltextIndexPlanner.java

```diff
@@ -308,12 +308,12 @@ public static void setUseActualEntryCount(boolean useActualEntryCount) {
             if (queryFilterPattern != null) {
                 if (ft != null && !queryFilterPattern.matcher(ft.toString()).find()) {
                     plan.addAdditionalMessage(Level.WARN, "Potentially improper use of index " + definition.getIndexPath() + " with queryFilterRegex "
-                            + queryFilterPattern + " to search for value " + ft.toString());
+                            + queryFilterPattern + " to search for value '" + ft.toString() + "'");
                 }
                  for (PropertyRestriction pr : filter.getPropertyRestrictions()) {
-                    if (!queryFilterPattern.matcher(pr.toString()).find()) {
+                    if (!pr.propertyName.startsWith(":") && !queryFilterPattern.matcher(pr.toString()).find()) {
                         plan.addAdditionalMessage(Level.WARN, "Potentially improper use of index " + definition.getIndexPath() + " with queryFilterRegex "
-                                + queryFilterPattern + " to search for value " + pr.toString());
+                                + queryFilterPattern + " to search for value '" + pr.toString() + "'");
```

Comment:
no need to call `.toString()`

```suggestion
                                + queryFilterPattern + " to search for value '" + pr + "'");
```